microsoft/testsuites/nested/common.py

```diff
@@ -94,6 +95,131 @@ def connect_nested_vm(
     return nested_vm
 
 
+def hyperv_connect_nested_vm(
+    host: RemoteNode,
+    guest_username: str,
+    guest_password: str,
+    port: int,
+    guest_image_url: str,
+    name: str = "L2-VM",
+    image_name: str = HYPERV_NESTED_VM_IMAGE_NAME,
+    switch_name: str = "nestedvmswitch",
+    nat_name: str = "nestedvmnat",
+) -> RemoteNode:
+    # Download nested vm image
+    image_folder_path = "C:\\Users\\lisatest"
+    file_path = f"{image_folder_path}\\{image_name}"
+    host.tools[WindowsFileManager].download_file(guest_image_url, file_path)
+
+    # setup NAT
+    host.tools[HyperV].setup_nat_networking(switch_name, nat_name)
+    host.tools[HyperV].create_vm(
+        name,
+        file_path,
+        switch_name,
+    )
+
+    # enable port forwarding
+    local_ip = host.tools[HyperV].get_ip_address(name)
+    host.tools[HyperV].setup_port_forwarding(nat_name, port, local_ip)
+
+    # setup connection to nested vm
+    nested_vm = RemoteNode(Node(name=name), 0, name)
+    nested_vm.set_connection_info(
+        public_address=host.public_address,
+        username=guest_username,
+        password=guest_password,
+        public_port=port,
+        port=port,
+    )
+
+    # wait for nested vm ssh connection to be ready
+    try_connect(
+        ConnectionInfo(
+            address=host.public_address,
+            port=port,
+            username=guest_username,
+            password=guest_password,
+        )
+    )
+
+    return nested_vm
+
+
+def hyperv_remove_nested_vm(
+    host: RemoteNode,
+    name: str = "L2-VM",
+    image_name: str = HYPERV_NESTED_VM_IMAGE_NAME,
+    switch_name: str = "nestedvmswitch",
+    nat_name: str = "nestedvmnat",
+) -> None:
+    # Delete nested vm image
+    image_folder_path = "C:\\Users\\lisatest"
+    file_path = f"{image_folder_path}\\{image_name}"
+    host.tools[WindowsFileManager].delete_file(file_path)
+
+    # Delete VM
+    host.tools[HyperV].delete_vm(name)
```

Comment:
Assign `host.tools[HyperV]` to a variable, it looks clearer.

Reply:
I assign it to a variable if it's used more than once in a function

Reply:
Yes, it should be, but it's written 3 times in this method.